src/Tests/UnitTest/ComGenerationTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿#if NET8_0_OR_GREATER
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Windows.ApplicationModel.DataTransfer.DragDrop.Core;
+using Windows.Graphics.Display;
+using Windows.Graphics.Printing;
+using Windows.Media;
+using Windows.Media.PlayTo;
+using Windows.Security.Authentication.Web.Core;
+using Windows.Security.Credentials;
+using Windows.Security.Credentials.UI;
+using Windows.UI.ApplicationSettings;
+using Windows.UI.Input;
+using Windows.UI.Input.Core;
+using Windows.UI.Input.Spatial;
+using Windows.UI.ViewManagement;
+using Xunit;
+using WinRT;
+using TestComponentCSharp;
+
+namespace UnitTest
+{
+    [GeneratedComInterface]
+    [Guid("15651B9F-6C6B-4CC0-944C-C7D7B0F36F81")]
+    internal partial interface IComInteropGenerated
+    {
+        Int64 ReturnWindowHandle(IntPtr hwnd, Guid iid);
+    }
+
+    public class ComGenerationTests
+    {
+        private static readonly Guid IID_IComInterop = new Guid("15651B9F-6C6B-4CC0-944C-C7D7B0F36F81");
+
+        [Fact]
+        public void TestHWND()
+        {
+            var comInterop = Class.ComInterop.As<IComInteropGenerated>();
```

Comment:
With the implementation you've done, this should be castable with just a regular cast. Something like the following instead of using `.As<T>()`.

```suggestion
            var comInterop = (IComInteropGenerated)(object)Class.ComInterop;
```

Reply:
I’m using `As<T>` here in order to match the pattern used by the built-in COM interop, but I suppose that should also work too.